### PR TITLE
Add phpstan with level 9

### PR DESCRIPTION
**Pull request by usox**
- Add new composer script entries for stan and qa (executes all
  qa-related scripts)
- Correct some typing issues

This also adds types for the internal Model properties. As strict typing
is enabled and the setter/getters are already types, adding types
without defaults should be safe.

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -38,7 +38,4 @@ jobs:
         run: composer install --prefer-dist --no-progress --no-suggest
 
       - name: Run Codestyle Check
```
Comment (WorksDev) [inline .github/workflows/php.yml:40]: ... QA ...

File: src/DeeplClient.php [hunk 2]
```diff
@@ -24,11 +24,11 @@
 
 class DeeplClient implements DeeplClientInterface
 {
-    private $deeplRequestFactory;
+    private DeeplRequestFactoryInterface $deeplRequestFactory;
```
Comment (WorksDev) [inline src/DeeplClient.php:27]: i think we can start with constructor property promotion, and removing php7.4 support

Comment (usox) [reply src/DeeplClient.php:27]: K. I'd prefer to update the contructors in a separate PR after this one to keep the currect one small and just stan related.

File: .github/workflows/php.yml (hunk 1 above)
Comment (usox) [reply .github/workflows/php.yml:40]: Right.

**Review by WorksDev — APPROVED** [review]

**Pull request merged**